esphome/components/t6615/t6615.cpp

```diff
@@ -0,0 +1,79 @@
+#include "t6615.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace t6615 {
+
+static const char *TAG = "t6615";
+static const uint8_t T6615_RESPONSE_BUFFER_LENGTH = 32;
+static const uint8_t T6615_HEADER[] = {0xFF, 0xFE};
+static const uint8_t T6615_COMMAND_GET_PPM[] = {0x02, 0x03};
+static const uint8_t T6615_COMMAND_GET_SERIAL[] = {0x02, 0x01};
+static const uint8_t T6615_COMMAND_GET_VERSION[] = {0x02, 0x0D};
+static const uint8_t T6615_COMMAND_GET_ELEVATION[] = {0x02, 0x0F};
+static const uint8_t T6615_COMMAND_GET_ABC[] = {0xB7, 0x00};
+static const uint8_t T6615_COMMAND_ENABLE_ABC[] = {0xB7, 0x01};
+static const uint8_t T6615_COMMAND_DISABLE_ABC[] = {0xB7, 0x02};
+static const uint8_t T6615_COMMAND_SET_ELEVATION[] = {0x03, 0x0F};
+
+void T6615Component::setup() {}
+
+void T6615Component::update() {
+  // CO2 sensor
+  if (this->co2_sensor_ != nullptr) {
```

Comment:
Rather than wrapping the entire function in an `if` statement you could use an early return and reduce the amount of indentation needed!
```suggestion
  if (this->co2_sensor_ == nullptr) {
      return;
  }
```

Reply:
The main reason I didn't make this an early-return was in case, in the future, it should publish any of the other values available to the sensor (elevation, sensor state, etc). I can change this and it could be changed back in the future though.

Reply:
It was only a suggestion 👍 If there are more values, I would put them each into a dedicated function and then call them from the `update()` function. Then each function would have it's own early return and it all still works.